codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/ViewServiceSourceGenerator.scala

```diff
@@ -75,213 +76,186 @@ object ViewServiceSourceGenerator {
-        |
-        |  /** Factory method of ${view.className} */
-        |  public static ${view.providerName} of(
-        |      Function<ViewCreationContext, ${view.className}> viewFactory) {
-        |    return new ${view.providerName}(viewFactory, "${view.viewId}", ViewOptions.defaults());
-        |  }
-        |
-        |  private ${view.providerName}(
-        |      Function<ViewCreationContext, ${view.className}> viewFactory,
-        |      String viewId,
-        |      ViewOptions options) {
-        |    this.viewFactory = viewFactory;
-        |    this.viewId = viewId;
-        |    this.options = options;
-        |  }
-        |
-        |  @Override
-        |  public String viewId() {
-        |    return viewId;
-        |  }
-        |
-        |  @Override
-        |  public final ViewOptions options() {
-        |    return options;
-        |  }
-        |
-        |  public final ${view.providerName} withOptions(ViewOptions options) {
-        |    return new ${view.providerName}(viewFactory, viewId, options);
-        |  }
-        |
-        |  /**
-        |   * Use a custom view identifier. By default, the viewId is the same as the proto service name.
-        |   * A different identifier can be needed when making rolling updates with changes to the view definition.
-        |   */
-        |  public ${view.providerName} withViewId(String viewId) {
-        |    return new ${view.providerName}(viewFactory, viewId, options);
-        |  }
-        |
-        |  @Override
-        |  public final Descriptors.ServiceDescriptor serviceDescriptor() {
-        |    return ${typeName(view.fqn.descriptorImport)}.getDescriptor().findServiceByName("${view.fqn.protoName}");
-        |  }
-        |
-        |  @Override
-        |  public final ${view.routerName} newRouter(ViewCreationContext context) {
-        |    return new ${view.routerName}(viewFactory.apply(context));
-        |  }
-        |
-        |  @Override
-        |  public final Descriptors.FileDescriptor[] additionalDescriptors() {
-        |    return new Descriptors.FileDescriptor[] {${view.fqn.parent.javaOuterClassname}.getDescriptor()};
-        |  }
-        |}
-        |""".stripMargin
+    JavaGeneratorUtils.generate(
+      packageName,
+      c"""|$managedComment
+          |
+          |public class ${view.providerName} implements $ViewProvider<${view.state.fqn}, ${view.impl}> {
+          |
+          |  private final $Function<$ViewCreationContext, ${view.className}> viewFactory;
+          |  private final String viewId;
+          |  private final $ViewOptions options;
+          |
+          |  /** Factory method of ${view.impl} */
+          |  public static ${view.providerName} of(
+          |      $Function<$ViewCreationContext, ${view.impl}> viewFactory) {
+          |    return new ${view.providerName}(viewFactory, "${view.viewId}", $ViewOptions.defaults());
+          |  }
+          |
+          |  private ${view.providerName}(
```

Comment:
Same for `view.providerName`->`view.provider`? It shouldn't lead to different output since this is the file where `view.provider` is defined, but I think it'd read a bit nicer and be 'good style' to prefer FQN's over strings where possible.